explore/src/main/scala/explore/observationtree/AsterismGroupObsList.scala

```diff
@@ -77,28 +81,25 @@ object AsterismGroupObsList {
           .fold(expanded - obsIds, expanded + obsIds)
       }
 
-    // TODO: We will need this when we start being able to drag targets.
-    // def parseDragId(dragId: String): Option[Either[Target.Id, Observation.Id]] =
-    //   Observation.Id
-    //     .parse(dragId)
-    //     .map(_.asRight)
-    //     .orElse(Target.Id.parse(dragId).map(_.asLeft))
-
-    def parseDragId(dragId: String): Option[Observation.Id] = Observation.Id.parse(dragId)
+    def parseDragId(dragId: String): Option[Either[Target.Id, Observation.Id]] =
+      Observation.Id
+        .parse(dragId)
+        .map(_.asRight)
+        .orElse(Target.Id.parse(dragId).map(_.asLeft))
 
     /**
-     * When we're dragging, we can either be dragging a single observation or a group of them. If we
-     * have a selection, and the drag id is part of the selection, we drag all the items in the
-     * selection. However, the user may have something selected, but be dragging something that is
-     * NOT in the selection - in which case we just drag the individual item.
+     * When we're dragging an observation, we can either be dragging a single observation or a group
+     * of them. If we have a selection, and the drag id is part of the selection, we drag all the
+     * items in the selection. However, the user may have something selected, but be dragging
+     * something that is NOT in the selection - in which case we just drag the individual item.
      */
-    def getDraggedIds(dragId: String, props: Props): Option[ObsIdSet] =
-      parseDragId(dragId).map { obsId =>
+    def getDraggedIds(dragId: String, props: Props): Option[Either[Target.Id, ObsIdSet]] =
+      parseDragId(dragId).map(_.map { obsId =>
```

Comment:
Would it make sense to use `Nested` here?

Reply:
Hmmm. Perhaps. I don't have a strong feeling for how Nested works, but I'll look into it.

Reply:
I don't think `Nested` is a good abstraction for here. Nested with an Either provides an easy way to map over the right side of the `Option[Either[]]`, at the expense of making the left more difficult. In this case, the `Either` is not being used in a `right biased` way. The fact that the observations ids are on the `Right` and I can use `map` to modify them is just a coincidence. Which makes me wonder if `Either` is even really the correct abstraction since it is now right-biased. A scala 3 union type would be better, I think. I'll change this to a match to better reflect the intentions.

Reply:
Maybe we can use a little ADT?